### ISSUE TITLE
Main bundle should only contain libraries needed for fresh environment load

### ISSUE THREAD

**Issue by outoftime** [popcodeorg/popcode#1292]
The main bundle should only contain the libraries needed to load a fresh environment, logged out, with no snapshot/gist ID in the URL and nothing in local storage. Everything else should be loaded via an async `import()`.

Specific things that I don’t think we need:

* Firebase database: we need auth to determine login state but if you’re logged out, that should be the end of it
* GitHub
* Anything related to validations (we are mostly good here but I think the custom HTML validation module unintentionally pulls some of its dependencies into the main bundle)
* The loop breaker: should be easy to load async now that `compileProject()` is async (and can skip it altogether if there is no JS)

We should also be deliberate about how we categorize various async bundles in the service worker configuration (e.g. I think the GitHub API should probably in a bundle classified as `optional` and loaded lazily) 

Probably other stuff too, those are just some big ones that come to mind.

**Comment by outoftime** [comment]
@ajgreenb I’m starting to assign issues to various milestones and I feel like #348 is too open-ended to ever really be “done”, so I figured I’d break this out as a more actionable version. Let me know your thoughts and if you’re down I will close #348

**Comment by outoftime** [comment]
(To be clear though console stuff is still a higher priority for now, just trying to get ducks aligned)

**Comment by ajgreenb** [comment]
I agree, I think making specific issues and using the `performance` label is the way to go here. We can break #348 apart and then create new issues as they arise.

**Comment by outoftime** [comment]
@ajgreenb cool—with this and #1293 I think we’ve covered the really actionable stuff from #348; other stuff is based on really ancient webpack stat analysis which I think is just worth doing more of in the future.
